npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb

```diff
@@ -715,7 +723,7 @@ def requirement_class
         end
 
         def version_regex
-          version_class::VERSION_PATTERN
+          Dependabot::NpmAndYarn::Version::VERSION_PATTERN
```

Comment:
I don't think this is _actually_ dynamic. And Sorbet requires constants to be static.